src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Validation/ValidatableObject.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Validation
+{
+    /// <summary>
+    /// Represents a base class for validating objects.
+    /// </summary>
+    public abstract class ValidatableObject : IAsyncValidatableObject
+    {
+        /// <inheritdoc/>
+        public virtual IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            var validationTask = ValidateAsync(validationContext, CancellationToken.None);
+
+            validationTask.Wait();
```

Comment:
While I commend the effort you've put into this interface, this still doesn't fix the issue. You shouldn't turn async code into sync, ever, no matter where you hide it, unless you're really-really forced to do that. As we've discussed under https://github.com/OrchardCMS/OrchardCore/pull/10934, there really is no such pressure here.